Source/player.cpp

```diff
@@ -1224,28 +1224,25 @@ bool SolidLoc(int x, int y)
 
 bool PlrDirOK(int pnum, int dir)
 {
-	int px, py;
 	bool isOk;
 
 	if ((DWORD)pnum >= MAX_PLRS) {
 		app_fatal("PlrDirOK: illegal player %i", pnum);
 	}
 	auto &player = plr[pnum];
 
-	px = player.position.tile.x + offset_x[dir];
-	py = player.position.tile.y + offset_y[dir];
-
-	if (px < 0 || !dPiece[px][py] || !PosOkPlayer(pnum, px, py)) {
+	Point position = player.position.tile + static_cast<Direction>(dir);

```

Comment:
Wouldn't it be better to redefine `PlrDirOK` as taking `Direction`?

Reply:
For sure. I didn't touch it yet as it was outside the scope of this one. The whole player file is getting massive refactoring in a next PR (already on the way).